browser/ui/views/brave_player/brave_player_action_icon_view.cc

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/brave_player/brave_player_action_icon_view.h"
+
+#include "base/strings/strcat.h"
+#include "brave/browser/ui/color/brave_color_id.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/vector_icons/vector_icons.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "brave/grit/brave_theme_resources.h"
+#include "chrome/browser/ui/browser_tabstrip.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/gfx/vector_icon_types.h"
+
+namespace {
+
+GURL GetPlayerURL(content::WebContents* web_contents) {
+  if (!web_contents) {
+    return {};
+  }
+
+  GURL url = web_contents->GetLastCommittedURL();
+  if (url.DomainIs("youtube.com") && url.has_path() && url.path() == "/watch" &&
+      url.has_query()) {
+    base::StringPairs key_value_pairs;
+    base::SplitStringIntoKeyValuePairs(url.query(), '=', '&', &key_value_pairs);
+    if (auto iter = base::ranges::find(key_value_pairs, "v",
+                                       &base::StringPairs::value_type::first);
+        iter != key_value_pairs.end()) {
+      return GURL(base::StrCat({kBravePlayerURL, "youtube/", iter->second}));
+    }
```

Comment:
How about using `net::GetValueForKeyInQuery()` simply?

Reply:
That's even better.